docs: Add paragraph about @ operator (#5822)
python-poetry/poetry: docs/dependency-specification.md

This will tell Poetry to install this version and this version only.
If other dependencies require a different version, the solver will ultimately fail and abort any install or update procedures.

### Multiple requirements

Multiple version requirements can also be separated with a comma, e.g. `>= 1.2, < 1.5`.

### `@` operator

> You can specify requirements, using `@` operator. This is understood the same way as `==` specifier, with one difference.

Reviewer: using the '@' operator, maybe

> After `@` symbol, you can use any specifiers that are valid in `pyproject.toml` file. For example:

Reviewer: After the '@' symbol, maybe.

Also, maybe best to stick with operator or symbol throughout.

```shell
poetry add django@^4.0.0
```

> Would translate to following entry in `pyproject.toml`:

Reviewer: Would translate through to the following entry in pyproject.toml

This will tell Poetry to install this version and this version only.
If other dependencies require a different version, the solver will ultimately fail and abort any install or update procedures.

### Using the `@` operator

> When adding dependencies via CLI `add` command, you can use `@` operator.

Reviewer 2: ```suggestion
When adding dependencies via CLI `add` command, you can use the `@` operator.
```

> This is understood the same way as `==` specifier, with one difference. After the `@` operator, you can use any
> specifiers that are valid in `pyproject.toml` file. For example:

Reviewer 2: ```suggestion
This is understood similarly to the `==` syntax, but also allows prefixing any
specifiers that are valid in `pyproject.toml`. For example:
```

```shell
poetry add django@^4.0.0
```

> Would translate to the following entry in `pyproject.toml`:

Reviewer 2: ```suggestion
The above would translate to the following entry in `pyproject.toml`:
```

```toml
Django = "^4.0.0"
```

> This operator, can also be used with in conjunction with word `latest`, to specify latest available version.
> For example:

Reviewer 2: ```suggestion
The special keyword `latest` is also understood by the `@` operator:
```

```shell
poetry add django@latest
```

> Would translate to the following entry in `pyproject.toml`:

Reviewer 2: ```suggestion
The above would translate to the following entry in `pyproject.toml`, assuming the latest release of `django` is `4.0.5`:
```

```toml
Django = "^4.0.5"
```

#### Extras

> To use extras for package using the `@` operator, formula is: `package[extra]@version`, for example:

Reviewer 2: ```suggestion
Extras and `@` can be combined as one might expect (`package[extra]@version`):
```

### Using the `@` operator

> When adding dependencies via CLI `add` command, you can use the `@` operator.

Reviewer 2: ```suggestion
When adding dependencies via `poetry add`, you can use the `@` operator.
```
